internal/controller/nonadminbackup_controller.go

```diff
@@ -213,27 +213,25 @@ func (r *NonAdminBackupReconciler) ValidateSpec(ctx context.Context, logrLogger
 	return false, false, nil
 }
 
-// UpdateSpecStatus updates the Spec and Status from the NonAdminBackup.
+// SyncVeleroBackupWithNonAdminBackup ensures the VeleroBackup associated with the given NonAdminBackup resource
+// is created, updated, and its status is reconciled. If the VeleroBackup does not exist, it creates a new one.
```

Comment:
This sentence is wrong, no? The function only creates VeleroBackup, update and status is reconcile is Velero responsibility

Would change

> is created, updated, and its status is reconciled. If the VeleroBackup does not exist, it creates a new one.

to 

> is created, if it does not exist.

Reply:
fixed.